rn/Teacher/ios/Submissions/PostGradesViewController.swift

```diff
@@ -0,0 +1,194 @@
+        case postTo, section
+    }
+
+    enum PostToVisisbility: String, CaseIterable {
+        case everyone, graded
+
+        var title: String {
+            switch self {
+            case .everyone:
+                return NSLocalizedString("Everyone", comment: "")
+            case .graded:
+                return NSLocalizedString("Graded", comment: "")
+            }
+        }
+
+        var subHeader: String {
+            switch self {
+            case .everyone:
+                return NSLocalizedString("Grades will be made visible to all students", comment: "")
+            case .graded:
+                return NSLocalizedString("Grades will be made visible to students with graded submissions", comment: "")
+            }
+        }
+    }
+    @IBOutlet weak var allGradesPostedView: UIView!
+    @IBOutlet weak var allGradesPostedLabel: DynamicLabel!
+    @IBOutlet weak var allGradesPostedSubheader: DynamicLabel!
+    private var sections: [String] = []
+    private var showSections: Bool = false
+    private var sectionToggles: [Bool] = []
+    private var gradesCurrentlyHidden = 0
+    private var visibility: PostToVisisbility = .everyone
+
+    static func create() -> PostGradesViewController {
+        let controller = loadFromStoryboard()
+        return controller
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setupSections()
+
+        postGradesButton.setTitle(NSLocalizedString("Post Grades", comment: ""), for: .normal)
+
+        allGradesPostedView.isHidden = true
+        allGradesPostedLabel.text = NSLocalizedString("All Posted", comment: "")
+        allGradesPostedSubheader.text = NSLocalizedString("All grades are currently posted.", comment: "")
+    }
+
+    func setupTableView() {
+        tableView.registerCell(SectionCell.self)
+        tableView.registerCell(PostToCell.self)
+    }
+
+    func setupSections() {
+        sections = ["section A", "section B"]     // FIXME: remove these static sections once we have real sections
+        sectionToggles = Array(repeating: false, count: sections.count)
+    }
+
+    @IBAction func actionUserDidClickPostGrades(_ sender: Any) {
+        print(#function)
+    }
+}
+
+extension PostGradesViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return Row.allCases.count +  (showSections ? sections.count : 0)
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        if indexPath.row == 0 {
+            cell = tableView.dequeue(for: indexPath) as PostToCell
+        } else {
+            cell = tableView.dequeue(for: indexPath) as SectionCell
+        }
+
+        if indexPath.row < Row.allCases.count { //  first two rows (Row enum)
+            if let row = Row(rawValue: indexPath.row) {
```

Comment:
Do you need both of these ifs? Or will the second one be enough? I _think_ it will be but I might be missing something.

Reply:
good point